analyzer/codechecker_analyzer/buildlog/log_parser.py

```diff
@@ -76,18 +76,22 @@
     '-fext-.*-literals',
     '-ffixed-r2',
     '-ffp$',
+    '-mfp16-format',
     '-fgcse-lm',
     '-fhoist-adjacent-loads',
     '-findirect-inlining',
     '-finline-limit',
     '-finline-local-initialisers',
     '-fipa-sra',
+    '-fmacro-prefix-map',
```

Comment:
Unfortunately, currently it is an "unwritten rule" that we support all versions of the analyzers. This is the concept of our checker labels too: we list all checkers ever existed in clang even with their old names if they were renamed. We have been following the same rule with this log parser: we skip flags if any clang version doesn't support them, but we have never removed any of these from the list just because Clang started to support it.
I agree that this is not a good approach. But I think this will not be solved nowadays. We were planning to create a 3pp tool that handles these compiler flags so that tool can also be used in other projects like CodeCompass. @gamesh411 started its development. I think this logic with different clang versions could be implemented in this tool if we get there in the future. As part of that project we could also review if any of these skipped flags are still unsupported by the latest clang.

Reply:
I am keeping this flag in the changeset unless unanimously advised otherwise.